app/src/common/shared/com/igalia/wolvic/ui/adapters/DownloadsAdapter.java

```diff
@@ -135,8 +137,42 @@ public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int
     public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
         DownloadItemViewHolder item = (DownloadItemViewHolder) holder;
         DownloadItemBinding binding = item.binding;
-        item.binding.setItem(mDownloadsList.get(position));
+        Download downloadItem = mDownloadsList.get(position);
+        item.binding.setItem(downloadItem);
         item.binding.setIsNarrow(mIsNarrowLayout);
+
+        switch (downloadItem.getStatus()) {
+            case Download.PENDING:
+                binding.thumbnail.setImageResource(R.drawable.ic_pending_circle);
+                break;
+            case Download.RUNNING:
+                binding.thumbnail.setImageResource(R.drawable.ic_downloading_circle);
+                break;
+            case Download.PAUSED:
+                binding.thumbnail.setImageResource(R.drawable.ic_pause_circle);
+                break;
+            case Download.FAILED:
+            case Download.UNAVAILABLE:
+                binding.thumbnail.setImageResource(R.drawable.ic_error_circle);
+                break;
+            case Download.SUCCESSFUL: {
+                if (downloadItem.getOutputFileUri() == null) {
+                    binding.thumbnail.setImageResource(R.drawable.ic_generic_file);
+                } else {
+                    final String itemUri = downloadItem.getOutputFileUri();
```

Comment:
You might create the variable before the if block because you check this in both branches.

Also I wonder when `getOutputFileUri()` returns null. Isn't that an error? Shouldn't we treat it as `Download.Unavailable`?

Reply:
Good suggestion! Fixed.